### PR TITLE
add cpu qos and mv nodeslo informer to states informer

### PR DESCRIPTION
**Pull request by zwzhang0107**
### Ⅰ. Describe what this PR does
#154

### CODE REVIEW

**Comment by jasonliu747** [comment]
@zwzhang0107 can you split this into separate PRs? one for CPU QoS, and the other for states_informer.

**Comment by zwzhang0107** [comment]
> 

it's already a separation of group identity hook implement(#154), since cpu qos definition dose not include much logic

File: pkg/util/config.go [hunk 1]
```diff
@@ -42,6 +42,27 @@ func DefaultResourceThresholdStrategy() *slov1alpha1.ResourceThresholdStrategy {
 	}
 }
 
+func DefaultCPUQOS(qos apiext.QoSClass) *slov1alpha1.CPUQoS {
```
Comment (jasonliu747) [inline pkg/util/config.go:45]: `DefaultCPUQoS`

**Comment by jasonliu747** [comment]
could you also fix #77 in this PR?

File: pkg/koordlet/statesinformer/register.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+func runCallbacks(objType reflect.Type, obj interface{}) {
+	callbacks, exist := stateUpdateCallbacks[objType]
+	if !exist {
+		klog.Errorf("states informer callbacks type %v not exist", objType)
+		return
+	}
+	for _, c := range callbacks {
+		klog.Infof("start running callback function %v for type %v", c.name, objType)
```
Comment (hormes) [inline pkg/koordlet/statesinformer/register.go:63]: Is this supposed to be a v(x) log ?

File: pkg/koordlet/statesinformer/states_informer.go [hunk 3]
```diff
@@ -19,39 +19,48 @@ package statesinformer
...
 	"k8s.io/klog/v2"
 
-	"github.com/koordinator-sh/koordinator/pkg/koordlet/metrics"
+	slov1alpha1 "github.com/koordinator-sh/koordinator/apis/slo/v1alpha1"
+	koordclientset "github.com/koordinator-sh/koordinator/pkg/client/clientset/versioned"
 	"github.com/koordinator-sh/koordinator/pkg/koordlet/pleg"
-	"github.com/koordinator-sh/koordinator/pkg/util"
 )
 
+var singleton StatesInformer
```
Comment (hormes) [inline pkg/koordlet/statesinformer/states_informer.go:40]: Could you avoid using singleton pattern?

File: pkg/util/config.go (hunk 1 above)
Comment (zwzhang0107) [reply pkg/util/config.go]: fixed

> 

done

**Comment by zwzhang0107** [comment]
> could you also fix #77 in this PR? Thanks

done

File: pkg/koordlet/statesinformer/register.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	description string
+	fn          UpdateCbFn
+}
+
+type UpdateCbFn func(s StatesInformer, statesObj interface{})
+
+func (s *statesInformer) RegisterCallbacks(objType reflect.Type, name, description string, callbackFn UpdateCbFn) {
+	callbacks, legal := s.stateUpdateCallbacks[objType]
+	if !legal {
+		panic(fmt.Sprintf("states informer callback register with type %v is illegal", objType))
```
Comment (saintube) [inline pkg/koordlet/statesinformer/register.go:37]: how about `klog.Fatalf()`?

Comment (zwzhang0107) [reply pkg/koordlet/statesinformer/register.go]: sure

**Comment by jasonliu747** [comment]
fixes #77

**Pull request merged**